### PR TITLE
New NovaPoshta Adapter 1.0.0

### PR DESCRIPTION
**Pull request by alexbakers**
## Description

NovaPoshta Adapter for tracking parcels.

## Steps to Test

```bash
$ yarn test packages/sources/novaposhta/test/unit/tracking.test.ts 
 PASS  packages/sources/novaposhta/test/unit/tracking.test.ts

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        1.672 s
```

```bash
$ yarn test packages/sources/novaposhta/test/integration/tracking.test.ts 
 PASS  packages/sources/novaposhta/test/integration/tracking.test.ts
{"level":"warn","time":1667520302342,"pid":88822,"hostname":"Alex-Baker","msg":"Could not load payload: [{\"instancePath\":\"\",\"schemaPath\":\"#/type\",\"keyword\":\"type\",\"params\":{\"type\":\"object\"},\"message\":\"must be object\"}]"}
{"level":"warn","time":1667520302342,"pid":88822,"hostname":"Alex-Baker","msg":"Falling back to default empty payload"}
{"level":"info","time":1667520302351,"pid":88822,"hostname":"Alex-Baker","msg":"Server listening on http://0.0.0.0:51933!"}

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   1 passed, 1 total
Time:        2.355 s, estimated 3 s
```

## Quality Assurance

- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [x] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)
- [x] All code changes have 100% unit and integration test coverage. If testing is not applicable or too difficult to justify doing, the reasoning should be documented explicitly in the PR.

### CODE REVIEW

File: packages/sources/novaposhta/src/adapter.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { Builder } from '@chainlink/ea-bootstrap'
+import {
+  Config,
+  ExecuteWithConfig,
+  ExecuteFactory,
+  AdapterRequest,
+  APIEndpoint,
+} from '@chainlink/ea-bootstrap'
+import { makeConfig } from './config'
+import * as endpoints from './endpoint' // The endpoints must be exported as shown in endpoint/index.ts for README generation.
```
Comment (boxhock) [inline packages/sources/novaposhta/src/adapter.ts:10]:
```suggestion
import * as endpoints from './endpoint'
```

File: packages/sources/novaposhta/src/config/index.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { Requester } from '@chainlink/ea-bootstrap'
+import { Config } from '@chainlink/ea-bootstrap'
+
+export const NAME = 'NOVAPOSHTA' // This should be filled in with a name corresponding to the data provider using UPPERCASE and _underscores_.
```
Comment (boxhock) [inline packages/sources/novaposhta/src/config/index.ts:4]:
```suggestion
export const NAME = 'NOVAPOSHTA'
```

File: packages/sources/novaposhta/src/config/limits.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "http": {
+    "NOVAPOSHTA": {
+      "rateLimit1s": 100
```
Comment (boxhock) [inline packages/sources/novaposhta/src/config/limits.json:4]: This seems quite high. Do you have a link to documentation of these limits?

File: packages/sources/novaposhta/src/endpoint/tracking.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    '"Documents" : [' +
+    '{' +
+    '"DocumentNumber": "' +
+    number +
+    '",' +
+    '"Phone": ""' +
+    '}' +
+    ']' +
+    '}' +
+    '}'
```
Comment (boxhock) [inline packages/sources/novaposhta/src/endpoint/tracking.ts:54]: We should build this as an object, then turn it into JSON with `JSON.stringify()`

File: packages/sources/novaposhta/test-payload.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+    "requests": [{
+        "number": "59000869676636"
+    }]
+}
+
```
Comment (boxhock) [inline packages/sources/novaposhta/test-payload.json:6]:
```suggestion

```

File: packages/sources/novaposhta/test/e2e/README.md [hunk 6]
```diff
@@ -0,0 +1 @@
+Tests that run the adapter and send an end to end request using live API endpoints.
```
Comment (boxhock) [inline packages/sources/novaposhta/test/e2e/README.md:1]: You can remove this file

**Review by boxhock — CHANGES_REQUESTED** [review]
Thanks for opening this PR! Referring to #2293 for the changeset issues

**Review by alexbakers — COMMENTED** [review]
@boxhock thank you very much, you showed in great detail what needs to be fixed.
I will take all the changes.
Please, if I merge branches incorrectly, let me know.

**Comment by mxiao-cll** [comment]
Please re-open if still relevant

**Pull request closed without merging**